slidingwindow.h

```diff
@@ -1744,5 +1744,176 @@ void ConvolutionInputGenerator_1D_parallel(
+ * \tparam Input_precision  	Number bits per pixel
+ * \tparam IFMDim_x          	Width of the Input Feature Map
+ * \tparam OFMDim_x           	Width of the Output Feature Map
+ * \tparam SIMD             	Number of input columns computed in parallel
+ * \tparam R          	  		Datatype for the resource used for FPGA implementation of the SWG  - safely deducible from the parameters
+ *
+ * \param in                	Input stream
+ * \param out               	Output stream
+ * \param numReps           	Number of time the function has to be repeatedly executed (e.g. number of images)
+ * \param r			  			Resource type for the hardware implementation of the memory block
+ */
+template<unsigned int ConvKernelDim_x,
+		 unsigned int IFMChannels,
+		 unsigned int Input_precision,
+		 unsigned int IFMDim_x,
+		 unsigned int OFMDim_x,
+		 unsigned int SIMD,
+		 typename R>
+void ConvolutionInputGenerator_1D_dws_lowbuffer(
+		stream<ap_uint<SIMD*Input_precision> > & in,
+		stream<ap_uint<SIMD*Input_precision> > & out,
+		const unsigned int numReps,
+		R const &r) {
+	CASSERT_DATAFLOW(IFMChannels % SIMD == 0);
+	const unsigned int multiplying_factor = IFMChannels/SIMD;
+	const unsigned int buffer_size = ConvKernelDim_x * multiplying_factor;
+	ap_uint<SIMD*Input_precision> inputBuf[buffer_size];
+#pragma HLS ARRAY_PARTITION variable=inputBuf complete dim=1
+	memory_resource(inputBuf, r);
+	const unsigned int cycles_read_block = multiplying_factor*(ConvKernelDim_x-1)-(ConvKernelDim_x-1);
+	const unsigned int baseIter = cycles_read_block + (OFMDim_x * buffer_size);
+	unsigned int inp = 0, index_write=0, index_read = 0, j = 0, internal_counter = 0;
+	for (unsigned int count_image = 0; count_image < numReps; count_image++) {
+		for (unsigned int i = 0; i < baseIter; i++) {
+			#pragma HLS PIPELINE II=1
+			if (inp < cycles_read_block) {// Initial buffer of ConvKernelDim lines
+				ap_uint<SIMD*Input_precision> inElem;
+				inElem = in.read();
+				inputBuf[inp] = inElem;
+				inp++;
+			}
+			else { // Read & write & update
+				// Read input buffer
+				if (inp < IFMDim_x*multiplying_factor){
+					if (inp < buffer_size || internal_counter >= buffer_size-multiplying_factor){
+						ap_uint<SIMD*Input_precision> inElem;
+						inElem = in.read();
+						index_write = inp % (buffer_size);
+						inputBuf[index_write] = inElem;
+						inp++;
+						internal_counter++;
+						if (internal_counter==buffer_size){
+							internal_counter=0;
+						}
+					}
+					else{
+						internal_counter++;
+					}
+				}
+
+				// Write output
+				ap_uint<SIMD*Input_precision> outElem = inputBuf[index_read];
+				out.write(outElem);
+
+				// Update read index pointer
+				if (j < ConvKernelDim_x-1){
+					index_read = index_read + multiplying_factor;
+					j++;
+				}
+				else{
+					index_read = index_read + (multiplying_factor+1);
+					j=0;
+				}
+				index_read = index_read%(buffer_size);
+
+				}
+		} // End base_iter
+	} // End count_image
+} // End generator
+
 
```

Comment:
The PR on the FINN side refers to `ConvolutionInputGenerator_1D_dilated_dws` but I couldn't see it in this file?

Reply:
Resolved as well :)